src/Components/Buttons.tsx

```diff
@@ -43,16 +44,20 @@ const linearGradientStyles = (props: LinearGradientStylesProps) => {
   return commonStyles;
 };
 
-const buttonContainerStyles = (props: ButtonContainerStylesProps) => ({
-  width: props.width || "100%",
-  height: props.height || 44,
-  borderRadius: props.borderRadius || 4,
-  borderWidth: props.borderWidth,
-  borderColor: props.borderColor,
-  marginVertical: props.marginVertical,
-  marginHorizontal: props.marginHorizontal,
-});
-
+const buttonContainerStyles = (props: ButtonProps) => {
+  if (props.style) {
+    return props.style;
+  }
+  return {
+    width: props.width || "100%",
+    height: props.height || 44,
+    borderRadius: props.borderRadius || 4,
+    borderWidth: props.borderWidth,
+    borderColor: props.borderColor,
+    marginVertical: props.marginVertical,
+    marginHorizontal: props.marginHorizontal,
```

Comment:
And if I want to pass a style with a `color` property? Why can not I do it as a consumer of this lib?

In react world it is usually done an easy way: there is some `defaultStyles` object, which is merged with user specified styles using a spread index.

The specification of style prop is usually following:

- Component should have its own styles
- User can specify additional styles
- User's styles should overwrite components' styles if there is a conflict